syz-manager/manager.go

```diff
@@ -1066,7 +1061,143 @@ func (mgr *Manager) fuzzerConnect() ([]rpctype.RPCInput, BugFrames, bool) {
+			return
+		}
+		mgr.modulesInitialized = true
+		mgr.coverFilterBitmap, mgr.coverFilter, err = mgr.updateCoverFilter()
+	})
+	if err != nil {
+		return nil, BugFrames{}, nil, err
+	}
+	r.CoverFilterBitmap = mgr.coverFilterBitmap
+	return corpus, BugFrames{memoryLeaks: memoryLeakFrames, dataRaces: dataRaceFrames}, mgr.coverFilter, nil
+}
+
+const EXT string = ".ko"
+
+func searchModuleName(data []byte) (string, error) {
+	key := []byte("name=")
+	pos := bytes.Index(data, key)
+	if pos == -1 {
+		return "", fmt.Errorf("not find name= keyword")
+	}
+	end := bytes.Index(data[pos:], []byte{0x00})
+	if end == -1 {
+		return "", fmt.Errorf("not find null char")
+	}
+	return string(data[pos+len(key) : pos+end]), nil
+}
+
+func getModuleName(path string) (string, error) {
+	file, err := elf.Open(path)
+	if err != nil {
+		return "", fmt.Errorf("failed to open %v", path)
+	}
+	section := file.Section(".modinfo")
+	if section == nil {
+		return "", fmt.Errorf("no .modinfo section")
+	}
+	data, err := section.Data()
+	if err != nil {
+		return "", fmt.Errorf("failed to read .modinfo")
+	}
+	name, err := searchModuleName(data)
+	if err != nil {
+		section = file.Section(".gnu.linkonce.this_module")
+		if section == nil {
+			return "", fmt.Errorf("no .gnu.linkonce.this_module section")
+		}
+		data, err = section.Data()
+		if err != nil {
+			fmt.Println("failed to read .gnu.linkonce.this_module")
+		}
+		name = string(data)
+	}
+	return name, nil
+}
+
+func findModulePaths(dir string) ([]string, error) {
+	files := []string{}
+	err := filepath.Walk(dir, func(path string, f os.FileInfo, err error) error {
+		if filepath.Ext(path) == EXT {
+			files = append(files, path)
+		}
+		return nil
+	})
+
+	return files, err
+}
+
+func autoFindModulePath(where, name string) (string, error) {
+	paths, err := findModulePaths(where)
+	if err != nil {
+		return "", err
+	}
+	for _, path := range paths {
+		if strings.TrimSuffix(filepath.Base(path), EXT) == name {
+			return path, nil
+		}
+	}
+	for _, slowpath := range paths {
+		mname, err := getModuleName(slowpath)
+		if err != nil {
+			return "", err
+		}
+		if mname == name {
+			return slowpath, nil
+		}
+	}
+	return "", fmt.Errorf("no match module path found")
+}
+
+func (mgr *Manager) getModulePath(name string) (string, error) {
+	modulePath := ""
+	var err error
+	if mgr.cfg.KernelObj != "" {
+		modulePath, err = autoFindModulePath(mgr.cfg.KernelObj, name)
+	}
+	for _, path := range mgr.cfg.ModuleObj {
+		modulePath, err = autoFindModulePath(path, name)
+		if err == nil {
+			return modulePath, nil
+		}
+	}
+	if err != nil {
+		return "", err
+	}
+	return modulePath, err
+}
+
+func (mgr *Manager) updateKernelModules(a *rpctype.ConnectArgs) error {
+	mgr.KernelModules = make(map[string]backend.KernelModule, len(a.Modules))
+	for _, rmodule := range a.Modules {
+		path, err := mgr.getModulePath(rmodule.Name)
+		if err != nil {
+			log.Logf(0, "failed to auto find module: %v", rmodule.Name)
```

Comment:
Also log the error here.
I think getModulePath can also return an empty path, which also means that we failed to find the module, right? It would be useful to log it as well, because currently there is no error, but you also won't see coverage for the module, and one may not understand why.
Also I would remove "auto" from the message, we don't have any other way to locate modules, so it looks excessive.